concordium-consensus/src/Concordium/Skov/Update.hs

```diff
@@ -479,87 +480,116 @@ doStoreBlock pb@GB.PendingBlock{..} = unlessShutDown $ do
-                NormalTransaction tx -> do
-                  focus <- getFocusBlock
-                  st <- blockState focus
-                  macct <- getAccount st $! transactionSender tx
-                  nextNonce <- fromMaybe minNonce <$> mapM (getAccountNonce . snd) macct
-                  -- If a transaction with this nonce has already been run by
-                  -- the focus block, then we do not need to add it to the
-                  -- pending transactions.  Otherwise, we do.
-                  when (nextNonce <= transactionNonce tx) $
-                      putPendingTransactions $! addPendingTransaction nextNonce WithMetadata{wmdData=tx,..} ptrs
-                CredentialDeployment _ ->
-                  putPendingTransactions $! addPendingDeployCredential wmdHash ptrs
-                ChainUpdate cu -> do
-                    focus <- getFocusBlock
-                    st <- blockState focus
-                    nextSN <- getNextUpdateSequenceNumber st (updateType (uiPayload cu))
-                    when (nextSN <= updateSeqNumber (uiHeader cu)) $
-                        putPendingTransactions $! addPendingUpdate nextSN cu ptrs
-              return (Just bi, ResultSuccess)
-          Duplicate tx -> return (Just tx, ResultDuplicate)
-          ObsoleteNonce -> return (Nothing, ResultStale)
+            ptrs <- getPendingTransactions
+            case wmdData of
+              NormalTransaction tx -> do
+                putPendingTransactions $! addPendingTransaction (transactionNonce tx) WithMetadata{wmdData=tx,..} ptrs
+              CredentialDeployment _ -> do
+                putPendingTransactions $! addPendingDeployCredential wmdHash ptrs
+              ChainUpdate cu -> do
+                putPendingTransactions $! addPendingUpdate (updateSeqNumber (uiHeader cu)) cu ptrs
+            -- The actual verification result here is only used if the transaction was received individually.
+            -- If the transaction was received as part of a block we don't use the result for anything.
+            -- todo: should we use it for e.g. logging?
+            return (Just bi, (mapTransactionVerificationResult verRes, Accepted))
+          Duplicate tx -> return (Just tx, (ResultDuplicate, Rejected))
+          ObsoleteNonce -> return (Nothing, (ResultStale, Rejected))
+      -- Reject transactions which definitely cannot be valid now or in a future block.
+      -- Transactions received individually must be verified successfully.
+      -- However transactions there is a looser requirement for transactions received
+      -- as part of a block. See the 'TransactionVerification' module for more details.
+      definitelyNotValid verificationResult =
+        case origin of
+          Single ->
+            case verificationResult of
+              TV.Ok _ -> False
+              _ -> True
+          Block ->
+            case verificationResult of
+              TV.NotOk _ -> True
+              _ -> False
```

Comment:
I would find these two more readable if all cases were spelled out. Seeing that there are only 3 it would not be too bad. And might help maintainability in the future.